cmd/controller-manager/app/controller-manager.go

```diff
@@ -174,19 +176,46 @@ func startControllers(opts *options.Options, stopChan <-chan struct{}) {
 	}
 }
 
-func setOptionsByFederationConfig(opts *options.Options) {
-	client := genericclient.NewForConfigOrDieWithUserAgent(opts.Config.KubeConfig, "federationconfig")
-
-	name := util.FederationConfigName
-	namespace := opts.Config.FederationNamespace
+func getFederationConfig(opts *options.Options) *corev1a1.FederationConfig {
 	fedConfig := &corev1a1.FederationConfig{}
-	err := client.Get(context.Background(), fedConfig, namespace, name)
+	if federationConfig == "" {
+		// there is no --federation-config specified, get `federation-v2` FederationConfig from the cluster
+		client := genericclient.NewForConfigOrDieWithUserAgent(opts.Config.KubeConfig, "federationconfig")
+
+		name := util.FederationConfigName
+		namespace := opts.Config.FederationNamespace
+		err := client.Get(context.Background(), fedConfig, namespace, name)
+		if err != nil {
+			glog.Infof("Cannot retrieve FederationConfig %q in namespace %q: %v. Default options are used.", name, namespace, err)
```

Comment:
nit: Please follow the convention of `<type name> "<qualified name>"` rather than `"<name>" in namespace "<namespace>"`